File: be/src/main/java/nl/zzpmatcher/userLogon/business/UserRegisterCommand.java
package nl.zzpmatcher.userLogon.business;

import lombok.Data;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;

@Data
public class UserRegisterCommand {

    private String emailaddress;
    private String password;
    private String password2;

    public User createUser() {

        final String encodedPassword = new BCryptPasswordEncoder().encode(password);

        return User.create(emailaddress, encodedPassword,"ROLE_USER");
    }
}

File: be/src/main/java/nl/zzpmatcher/userlogon/business/UserRegisterCommand.kt
package nl.zzpmatcher.userlogon.business

import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder

class UserRegisterCommand {

    var emailaddress: String? = null
    var password: String? = null
    var password2: String? = null

    fun createUser(): User {

        val encodedPassword = BCryptPasswordEncoder().encode(password!!)

        return User.create(emailaddress!!, encodedPassword, "ROLE_USER")
    }
}
